### PR TITLE
Quelch deprecation warning in tests

### PR DESCRIPTION
**Pull request by uranusjr**
Re-submitting #20742 with conflicts removed. Most significantly, `mark_tasks` is modified to not use bare execution dates, but AIP-39 data intervals.

### CODE REVIEW

File: airflow/api_connexion/endpoints/role_and_permission_endpoint.py [hunk 1]
```diff
@@ -147,7 +147,7 @@ def post_role() -> APIResponse:
     if not role:
         perms = [(item['action']['name'], item['resource']['name']) for item in data['permissions'] if item]
         _check_action_and_resource(security_manager, perms)
-        security_manager.init_role(role_name=data['name'], perms=perms)
+        security_manager.bulk_sync_roles([{"role": data["name"], "perms": perms}])
```
Comment (ephraimbuddy) [inline airflow/api_connexion/endpoints/role_and_permission_endpoint.py:150]: Seems unrelated?

File: airflow/api_connexion/schemas/dag_schema.py [hunk 2]
```diff
@@ -79,7 +79,7 @@ class DAGDetailSchema(DAGSchema):
     timezone = TimezoneField()
     catchup = fields.Boolean()
     orientation = fields.String()
-    concurrency = fields.Integer()  # TODO: Remove in Airflow 3.0
+    concurrency = fields.Method("get_concurrency")  # TODO: Remove in Airflow 3.0
```
Comment (ephraimbuddy) [inline airflow/api_connexion/schemas/dag_schema.py:82]: The concurrency here is the same as `max_active_tasks` but deprecated. See https://github.com/apache/airflow/blob/fce46e603b742087a4b72aad42b3126b9c262aa1/airflow/models/dag.py#L388-L396

**Review by ephraimbuddy — COMMENTED** [review]
I think I have seen the role and permission endpoint failing at times but when you use --with-db-init, it runs well. Maybe we should address it in a separate PR?

File: airflow/api_connexion/endpoints/role_and_permission_endpoint.py (hunk 1 above)
Comment (uranusjr) [reply airflow/api_connexion/endpoints/role_and_permission_endpoint.py:150]: `init_role()` is deprecated (and simply calls `bulk_sync_roles`), this removes one deprecation warning.

File: airflow/api_connexion/schemas/dag_schema.py (hunk 2 above)
Comment (uranusjr) [reply airflow/api_connexion/schemas/dag_schema.py:82]: Yeah that’s the point. `concurrency = fields.Integer()` triggers Marshmallow to access the deprecated `concurrency` property and emits a deprecation warning, and this change instead accesses `max_active_tasks` to avoid it. The additional `get_concurrency` method is added because Marshmallow is too smart to detect that `max_active_tasks` is already declared on the schema (right below this line) and throws an exception if we also directly access the same attribute for this field.

Comment (ephraimbuddy) [reply airflow/api_connexion/schemas/dag_schema.py:82]: Ok. 
You mean that in `get_concurrency` limit, if we use `obj.max_active_tasks` instead of `obj.max_active_runs`  it'll cause a problem?

Comment (uranusjr) [reply airflow/api_connexion/schemas/dag_schema.py:82]: Oh… no, I meant if we use `concurrency = fields.Integer("max_active_tasks")` it’ll error. The `max_active_runs` below is a typo 🤦

File: airflow/api_connexion/schemas/dag_schema.py [hunk 3]
```diff
@@ -90,6 +90,10 @@ class DAGDetailSchema(DAGSchema):
     is_paused = fields.Method("get_is_paused", dump_only=True)
     is_active = fields.Method("get_is_active", dump_only=True)
 
+    @staticmethod
+    def get_concurrency(obj: DAG):
+        return obj.max_active_runs
```
Comment (uranusjr) [inline airflow/api_connexion/schemas/dag_schema.py]:
```suggestion
        return obj.max_active_tasks
```

File: airflow/api_connexion/schemas/dag_schema.py (hunk 2 above)
Comment (uranusjr) [reply airflow/api_connexion/schemas/dag_schema.py:82]: fixed

**Review by ephraimbuddy — APPROVED** [review]
